x-pack/plugins/apm/server/index.ts

```diff
@@ -69,6 +69,80 @@ const configSchema = schema.object({
     schema.string({ defaultValue: '' }),
     schema.never()
   ),
+  featureFlags: schema.object({
+    agentConfigurationAvailable: schema.conditional(
+      schema.contextRef('serverless'),
+      true,
+      schema.boolean({
+        defaultValue: false,
+      }),
+      schema.oneOf([schema.literal(true)], { defaultValue: true })
```

Comment:
I didn't understand the notEqual logic here.

Basically, if we are not running in serverless context, its default value could be `true`?

Reply:
Yes, i understand the schema.conditional.
What i don't get it is, when we are not running in serverless mode, you always want this settings to be true ? This is what i don't get it

Reply:
Reason for asking this is, since you are setting it as true in non serverless mode, you will have to update the security teams tests also where they check for exposed settings.

I am pretty sure, they will also ask for documentation update for these exposed settings.

Reply:
> If we are not running in serverless mode, it will always be true.

> is this the expected behaviour ?

Yes, we want these values to be true when we are not in serverless

> This might also need documentation update for these exposed settings. The moment you update the security tests, someone from the security team will provide more details

As you said I needed to update the security test and the team was ping to review these changes, I'm waiting for them for more details